src/compiler/resolutionCache.ts

```diff
@@ -281,6 +285,10 @@ namespace ts {
             return !!value && !!value.length;
         }
 
+        function isInvalidatedAutomaticTypeReferenceDirective(path: Path): boolean {
+            return hasChangedTypingsFiles && endsWith(path, inferredTypesContainingFile);

```

Comment:
Isnt this exactly same as `hasChangedAutomaticTypeDirectiveNames`

Reply:
It’s called from `resolveNamesWithLocalCache` so we need to know that it was an automatic type reference directive (hence checking the path). If it were just `return hasChangedAutomaticTypeDirectiveNames`, all imports would appear invalidated too.

Reply:
Oh, you mean is `hasChangedTypingsFiles` the same as `hasChangedAutomaticTypeDirectiveNames`? No, `hasChangedAutomaticTypeDirectiveNames` can be true when `hasChangedTypingsFiles` is false.

Reply:
oh i see .. you need this to ensure that file is invalidated and re-resolved.